### ISSUE TITLE
FileNotFoundError in hapmultisequencer while removing rules_filename

### ISSUE THREAD

**Issue by stscijgbot-hstdp** [spacetelescope/drizzlepac#1060]
_Issue [HLA-582](https://jira.stsci.edu/browse/HLA-582) was created on JIRA by [Robert Swaters](https://jira.stsci.edu/secure/ViewProfile.jspa?name=rswaters):_

```
2021180172021 INFO src=drizzlepac.hapmultisequencer- Updating these drizzle products for CAOM compatibility:
2021180172021 INFO src=drizzlepac.hapmultisequencer-     hst_skycell-p2381x06y09_wfc3_uvis_f547m_fine-all-all_drc.fits
2021180172031 INFO src=drizzlepac.hapmultisequencer-     hst_skycell-p2381x06y09_wfc3_uvis_f657n_fine-all-all_drc.fits
2021180172043 INFO src=drizzlepac.hapmultisequencer- Removed rules file hst_skycell-p0797x13y05_acs_wfc_f435w_fine-all-all_j8ph03iz_header_hap.rules
^G^G^G
Traceback (most recent call last):
  File "/ifs/pipeline/dms/rhel7/pkgs/miniconda3/envs/caldp_master/lib/python3.7/site-packages/drizzlepac/hapmultisequencer.py", line 245, in run_mvm_processing
    driz_list = create_drizzle_products(total_obj_list)
  File "/ifs/pipeline/dms/rhel7/pkgs/miniconda3/envs/caldp_master/lib/python3.7/site-packages/drizzlepac/hapmultisequencer.py", line 129, in create_drizzle_products
    os.remove(rules_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'hst_skycell-p0797x13y05_acs_wfc_f435w_fine-all-all_j8ph03iz_header_hap.rules'
```
This happens for the same dataset that was the subject of HLA-580

**Comment by stscijgbot-hstdp** [comment]
_Comment by [Michele De La Pena](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mdelapena) on [JIRA](https://jira.stsci.edu/browse/HLA-582?focusedCommentId=535818#comment-535818):_

Did you capture the output error properly?   The update for for CAOM is for a wfc3/uvis data set, but the rules file issue is for acs/wfc.  Was this processing for a more interesting dataset than the one you originally gave me (icnk*)?

**Comment by stscijgbot-hstdp** [comment]
_Comment by [Michele De La Pena](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mdelapena) on [JIRA](https://jira.stsci.edu/browse/HLA-582?focusedCommentId=535819#comment-535819):_

FYI - When I processed icnk* dataset (p2381x06y09), I was successful, and there were no issues removing the rules files.

**Comment by stscijgbot-hstdp** [comment]
_Comment by [Robert Swaters](https://jira.stsci.edu/secure/ViewProfile.jspa?name=rswaters) on [JIRA](https://jira.stsci.edu/browse/HLA-582?focusedCommentId=535875#comment-535875):_

I did capture the output error correctly, but did not look at it closely.  It is, as you called it, more interesting, because the error is for hst_skycell-p0797x13y05_acs_wfc_f435w_fine-all-all_j8ph03iz_header_hap.rules, whereas the dataset being processed is skycell-p2381x06y09. This suggests the underlying problem is related to parallel processing. Grepping through all the logs indicates that the removal of the rules files is rather enthusiastic. In the log for skycell-p0797x13y05 a rules files for different, unrelated datasets are being removed as well, pulling the rug underneath from later processes that then also try to remove them.
```
ALOG_1624986965_HAP_MultiVisitMosaic_skycell-p0797x13y05.out:2021180173300 INFO src=drizzlepac.hapmultisequencer- Removed rules file hst_skycell-p2381x05y09_wfc3_uvis_f547m_fine-all-all_icnk20ay_header_hap.rules
ALOG_1624986965_HAP_MultiVisitMosaic_skycell-p0797x13y05.out:2021180173300 INFO src=drizzlepac.hapmultisequencer- Removed rules file hst_skycell-p0797x13y05_acs_wfc_f814w_fine-all-all_j8ph03iv_header_hap.rules
ALOG_1624986965_HAP_MultiVisitMosaic_skycell-p0797x13y05.out:2021180173300 INFO src=drizzlepac.hapmultisequencer- Removed rules file hst_skycell-p0797x13y05_acs_wfc_f555w_fine-all-all_j8ph03j3_header_hap.rules
ALOG_1624986965_HAP_MultiVisitMosaic_skycell-p0797x13y05.out:2021180173300 INFO src=drizzlepac.hapmultisequencer- Removed rules file hst_skycell-p0797x13y05_acs_wfc_f555w_fine-all-all_j8ph03j1_header_hap.rules
ALOG_1624986965_HAP_MultiVisitMosaic_skycell-p0797x13y05.out:2021180173300 INFO src=drizzlepac.hapmultisequencer- Removed rules file hst_skycell-p2381x06y09_wfc3_uvis_f657n_fine-all-all_icnk20a8_header_hap.rules
```

**Comment by stscijgbot-hstdp** [comment]
_Comment by [Michele De La Pena](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mdelapena) on [JIRA](https://jira.stsci.edu/browse/HLA-582?focusedCommentId=535952#comment-535952):_

Changed the way the skycell filenames are collected in order to generate the Rules file list. Modified the code from using 'glob' to collecting the filenames from the exposure object attribute as is done for SVM processing.  This ensures only filenames which belong to the specific processing instance are accommodated. 

The fix is out for review Git PR#&#x2060;1061.

**Comment by stscijgbot-hstdp** [comment]
_Comment by [Michele De La Pena](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mdelapena) on [JIRA](https://jira.stsci.edu/browse/HLA-582?focusedCommentId=536122#comment-536122):_

The GitHub PR#&#x2060;1061 has been approved and merged.

**Comment by stsci-hack** [comment]
Closed by #1061
